### PR TITLE
INSTALL.md: Update Install hints

### PR DESCRIPTION
**Pull request by ax3l**
- update an outdated boost install line
- update gcc recommendation to 4.9
- C++11 hints
- adios 1.10.0 was released (ships mxml)

### CODE REVIEW

**Comment by PrometheusPi** [comment]
looks good to me, all links are correct :+1:

File: INSTALL.md [hunk 1]
```diff
@@ -153,10 +156,10 @@ Some of our examples will also need **libSplash**.
     - example:
       - `mkdir -p ~/src ~/build ~/lib`
       - `cd ~/src`
-      - `wget http://users.nccs.gov/~pnorbert/adios-1.9.0.tar.gz`
-      - `tar -xvzf adios-1.9.0.tar.gz`
-      - `cd adios-1.9.0`
-      - `CFLAGS="-fPIC" ./configure --enable-static --enable-shared --prefix=$HOME/lib/adios-1.9.0 --with-mxml=$MXML_ROOT --with-mpi=$MPI_ROOT --with-zlib=/usr`
```
Comment (PrometheusPi) [inline INSTALL.md]: @ax3l Why did you remove `http://users.nccs.gov/~pnorbert/adios-1.10.0.tar.gz`?

Comment (ax3l) [inline INSTALL.md]: see commit and PR description

Comment (PrometheusPi) [inline INSTALL.md]: thank you
